### ISSUE TITLE
work on iOS or Android?

### ISSUE THREAD

**Issue by mnorst** [wjakob/nanogui#276]
thanks

**Comment by prsolucoes** [comment]
I think no, because it was made for OpenGL 3.

**Comment by svenevs** [comment]
@prsolucoes is correct, OpenGL 3.x or 4.x are required.  There was [some work for GLES put forward](https://github.com/wjakob/nanogui/pull/175) a while ago, but that is now stale.  Supporting OpenGL 2.x is technically possible, but would require significant effort to implement.

**Comment by cesss** [comment]
Does nanogui call OpenGL 3.x directly? Doesn't it make the calls indirectly through nanovg? If that's the case, I read that nanovg has several backends, and not only OpenGL 3.x... it seems 2.x is also supported, and even some ES versions...

**Comment by svenevs** [comment]
> Does nanogui call OpenGL 3.x directly?

Yes, see [`glutil.cpp`](https://github.com/wjakob/nanogui/blob/master/src/glutil.cpp) (specifically e.g., renderbuffer stuff).

I know that the future version of NanoGUI (which is not fully released yet) has GLES support, but that will not be back-ported here.  In the interim, I suggest you make a fork of NanoGUI and change / remove things as needed.  If you are successful, I'm sure others would be grateful to you for sharing (e.g., share a link here).

As far as I understand it, this repository will never have GLES support at this point (since it is already solved in the future version).
